plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConversionVisitor.kt

```diff
@@ -55,23 +55,19 @@ class StmtConversionVisitor : FirVisitor<Exp, StmtConversionContext>() {
         val expr = returnExpression.result.accept(this, data)
         // TODO: respect return-based control flow
         val returnVar = data.signature.returnVar
-        data.addStatement(Stmt.LocalVarAssign(returnVar.toLocalVar(), expr))
+        data.addStatement(Stmt.LocalVarAssign(returnVar.toLocalVar(), expr.withType(returnVar.viperType)))
```

Comment:
Perhaps an extension method on `Exp` that takes `TypeEmbedding` would be good?

Reply:
Do you mean an extension function that let's you write `expr.withType(returnVar.type)`? I don't really see the benefit over `expr.withType(returnVar.viperType)`.

Reply:
Yeah.  I guess the main advantage would be that we don't have to think about type extensions vs types here, but it's not a big deal.